### PR TITLE
New RFP Ideas for Wave 3

### PR DESCRIPTION
**Pull request by DecentralTech**
# RFP: New RFP Ideas for Wave 3

**Name of Project:** Simple Storage webapps

**RFP Category:** `app-dev`

**Proposer:** [@DecentralTech]

## Project Description

Single page apps that demonstrate using Filecoin to store and retrieve files. Example workflow: drag + drop a file, pay to store it, retrieve it. Show miner info, CIDs.
One version could eli5 (explain-it-like-I'm-5) to new users as a tutorial or tour.
Your proposal should provide examples of applications and interfaces your team has built and links to their code repos.
Also see  **Filecoin community resources**  for an initial list of Filecoin developer tools.

## Outcome

- File network disk, refer to the IPFS WebUI demo, without registering the login system, allowing users to upload files, view uploaded files, and obtain files
- The function relies on the filecoin chain to interact with the chain instantly

## Team

| Job | staffing | Period | Tasks |
| --- | --- | --- | --- |
| Technical Director | 1 person | 0.25 month | overall scheduling |
| Frontend Engineer | 1 person | 2 month | frontend coding |
| Backend Engineer | 1 person | 1.5 month | backend coding |
| Blockchain Engineer | 1 person | 0.5 month | chain data interact |
| UI Designer | 1 person | 0.5 month | UI design |

## Upgrade Plans

- User system
- Mobile APP

**Name of Project:** Filecoin Chain State Explorer

**RFP Category:** `app-dev`

## Project Description

An interactive visual tool to dive into the Filecoin state trie would help protocol developers and researchers do in-depth analysis. The 1st level could be an interpretation of the actual tree and include key Actors, a map of the HAMT data structure, etc. Currently a latest trie snapshot is ~10GB.
Also see the RFP for Diagram the Filecoin blockchain

Existing tools:
- The lotus node implementation of Filecoin has a Postgres database tool called chainwatch on top of which a visual chain explorer has been built.
- Several block explorers for Filecoin (_filscan.io, filscout.io, filplorer.com_) have been developed
- Filecoin’s state trie is unique but this  _Ethereum state trie diagram_  provides an example of what may be involved
Preferred applicants should be able to work independently and learn how the Filecoin state trie is structured using the  _Filecoin Specification_  and the various Filecoin node implementations.

## Outcome

- Interactive and high performance chain data tree explorer
- Along with updated **filplorer.com**

## Team

(**Filplorer.com** was built by our team previously, chain state explorer is planned to updated upon **filplorer.com**)

| Job | staffing | Period | Tasks |
| --- | --- | --- | --- |
| Technical Director | 1 person | 0.5 month | overall scheduling |
| Frontend Engineer | 1 person | 1 month | frontend coding |
| Backend Engineer | 1 person | 1 month | backend coding |
| Blockchain Engineer | 1 person | 0.5 month | chain data interact |
| UI Designer | 1 person | 0.5 month | UI design |

**Name of Project:** Diagram the Filecoin Blockchain

**RFP Category:** `app-dev`

## Project Description

Diagrams of the Filecoin blockchain at the level of detail and visual quality of Lee Thomas’  diagram of the _Ethereum Yellow paper  and  state trie_  [ Source ].
Preferred applicants should be able to work independently and learn how the Filecoin blockchain is structured using the  _Filecoin Specification_  and the various Filecoin node implementations.

## Outcome

- Very clear diagram explanation of Filecoin blockchain
- Maintenance of long-term update of technical specifications

## Team

| Job | staffing | Period | Tasks |
| --- | --- | --- | --- |
| Technical Director | 1 person | 0.25 month | overall scheduling |
| Blockchain Engineer | 1 person | 0.5 month | chain explanation |
| Backend Engineer | 1 person | 0.5 month | bdrawing |

**Name of Project:** Web archiving browser extension or Twitter bot

## Project Description

A browser extension that allows users to archive websites to IPFS and backup to Filecoin. This extension could also provide a Filecoin wallet so that users can pay for archival and as a way to keep track of what they have saved.
In the future this could be expanded to ways to collaborate on and share archived bookmarks in online communities or as bookmarking services (e.g.  _Pinboard_ , Pinterest, etc.)
A Twitter bot such as the former  _Link Archiver_  could automatically archive websites from tweeted links.
This project would need to explore the required infrastructure for these services.

## Outcome

- Current page content generate html files and save it on IPFS. Filecoin Deals was optional for users
- .No Filecoin Wallet included but wallet API set aside, because found some other team have proposed Filecoin Wallet
- Chrome and Firefox extension

## Team

| Job | staffing | Period | Tasks |
| --- | --- | --- | --- |
| Technical Director | 1 person | 0.1 month | overall scheduling |
| Backend Engineer | 1 person | 0.5 month | backend coding |
| Blockchain Engineer | 1 person | 0.5 month | chain data interact |
| UI Designer | 1 person | 0.1 month | UI design |

## Overall Summary

| Job | staffing | Period | 
| --- | --- | --- |
| Technical Director | 1 person | 1.1 month | 
| Frontend Engineer | 1 person | 3 month | 
| Backend Engineer | 1 person | 3 month | 
| Blockchain Engineer | 1 person | 1.75 month | 
| UI Designer | 1 person | 1.6 month | 

| Budget | Amount | 
| --- | --- | 
| Simple Storage webapps | $19,500  |
| Filecoin Chain State Explorer | $14,000  |
| Diagram the Filecoin Blockchain | $4,000  |
| Web archiving browser extension or Twitter bot | $4,800 |

## Contact Info

E-mail:chenwang@decentral.cc

## Team Website

https://dianduidian.tech/

### Relevant Experience
The team are all distributed storage professional researchers，all members be devoted to distributed storage promotion and equipment development，Distributed storage application development and data center integration construction.
Team members actively exploring and solving the problems of supply chain finance, traceability and bill storage.
We will use blockchain technology to achieve application innovation and strive to empower the real economy.

We finished a lot of projects that are Blockchain-related：
- Dianduidian-application
- Blockchain of XiAn education
- Blockchain of GuangZhou procuratorate
- Blockchain revolution of ShangHai antique market
- IPFS-browser

The famous IPFs community：
- https://ipfsdrop.com/

### CODE REVIEW

**Comment by eshon** [comment]
Hi @DecentralTech,

Are you applying for 4 different projects in this proposal? 

I don't see any technical detail about any of the projects beyond what we wrote in the RFPs. Please describe in more detail what your team is planning to build and how with as much technical detail as possible.

**Comment by DecentralTech** [comment]
Hello @eshon 

Here's a schematic of the technical logic of Web archiving browser extension. And we have implemented the code for Archive Website to IPFS part.

awfe

Here is a 30 sec video demo: [30 sec video](https://cdn.dianduidian.tech/video/demo-Archive-Website-to-Filecoin_1.mp4
) Archive a website to IPFS and MFS, output the hash, download the website file from IPFS and it can be viewed with css styling offline.

Is this information sufficient for the devgrant application for the browser extension project? If this information is OK, we would be happy to have our product manager put together a schematic presentation of the other 3 project.

**Comment by ErinOCon** [comment]
Archived
